src/librustc_trans/debuginfo/metadata.rs

```diff
@@ -1569,7 +1568,9 @@ fn prepare_enum_metadata<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
         match cached_discriminant_type_metadata {
             Some(discriminant_type_metadata) => discriminant_type_metadata,
             None => {
-                let discriminant_llvm_type = adt::ll_inttype(cx, inttype);
+                let discriminant_llvm_type = Type::from_integer(cx,
```

Comment:
It will be when I'm done, but it also needs to be recoded to deal with `layout::Integer`.